lib/infer/membership.js

```diff
@@ -77,6 +116,32 @@ function countModuleIdentifiers(comment, identifiers) {
 }
 
 /**
+ * Returns the comment object after normalizing Foo.prototype and Foo# expressions
+ * @param {Object} comment parsed comment
+ * @returns {Object} the normalized comment
+ */
+function normalizeMemberof(comment) {
+  var memberof = comment.memberof;
+  var index = memberof.lastIndexOf('.');
+
+  if (index !== -1 && memberof.slice(index + 1) === 'prototype') {
```

Comment:
I don't know how important performance is to you. It surely would result in more compact code.  But the next line would read `comment.memberof = memberof.slice(0, -10);` which looks a little bit to much on the magical side for my taste.

I could also use a capturing regexp: `memberof.match(/^(.*).prototype$/)` and use match[1] like this:

```js
var match = memberof.match(/^(.*).prototype$/);

if (match !== null) {
  comment.memberof = match[1];
```

I'm not sure if that's an improvement in the end. But if this works for you:

```js
  if (memberof.match(/.prototype$/) !== null) {
    comment.memberof = memberof.slice(0, -10);
    comment.scope = 'instance';

    return comment;
  }
```

I am totally on board.